LanguageFeatures/Patterns/shared_case_scope_A02_t02.dart

```diff
@@ -38,24 +38,26 @@
 // SharedOptions=--enable-experiment=patterns
 
 import "../../Utils/expect.dart";
+import "../../Utils/static_type_helper.dart";
 
-String test1(int x) {
+String test1(num x) {
   switch (x) {
-    case final a when a == 42:
-    case final a when a == 0:
+    case final a when a is int:
     case final int a:
+      a.expectStaticType<Exactly<int>>();
       print(a);
       return "match";
     default:
       return "default";
   }
 }
 
-String test2(int x) {
+String test2(num x) {
   switch (x) {
-    case var a when a == 42:
+    case var a when a is int:
     case int a when a == 0:
-    case var a:
+    case var a when a is Never: // this breaks the promotion
```

Comment:
I'm not sure about that comment: The flow analysis of this branch would presumably mark it as stopped: There is no way to continue executing the program such that the previous step was a successful `is Never` test. This just means that it doesn't contribute to the merge of the branches, and that again means that `a` is promoted by the other branches and this one is ignored. So `a` is promoted to `int`.

That doesn't sound like 'breaks the promotion' ... ?